pkg/deploy/controller/deployment/controller.go

```diff
@@ -59,6 +59,9 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 		// transition to failed immediately.
 		if deployutil.IsDeploymentCancelled(deployment) {
 			nextStatus = deployapi.DeploymentStatusFailed
+			if err := c.cancelDeployerPods(deployment); err != nil {
+				return err
```

Comment:
Interesting q... I do think that's a larger investigation. Pretty sure this patch will do what we want for now.